### PR TITLE
Adding mdbook-based user manual

### LINKED ISSUE #428: Migrate docs to use rustdoc books
Migrate the existing Jekyll docs over to using Rustdoc books to avoid the overhead of managing Jekyll.

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR fixes #428 by removing Jekyll in favor of using mdbook to generate documentation for Stabilizer.

Sample overview page:
image

TODO:
 - Verify release CI

### CODE REVIEW

File: book/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+stabilizer-manual
+src/firmware
```
Comment (jordens) [inline book/.gitignore:2]:
```suggestion
/stabilizer-manual
/src/firmware
```

I think.

**Review by jordens — APPROVED** [review]
Cool!

File: book/.gitignore (hunk 1 above)
Comment (ryan-summers) [reply book/.gitignore]: Also needed a trailing slash - thanks for catching this one.

**Comment by ryan-summers** [comment]
bors merge

**Pull request merged**